theseus/core/tests/test_objective.py

```diff
@@ -280,6 +280,26 @@ def _check_error_for_data(v1_data_, v2_data_, error_, error_norm_2_):
         assert error.shape == (batch_size, 2 * dof)
         _check_error_for_data(v1_data, v2_data, error, error_norm_2)
 
+        # For input_data and also_update
+
+        v1_data = torch.ones(batch_size, dof) * f1 * 0.1
+        v2_data = torch.ones(batch_size, dof) * f2 * 0.1
+
+        input_data = {"v1": v1_data, "v2": v2_data}
+        error = objective.error(input_data=input_data, also_update=True)
+        error_norm_2 = objective.error_squared_norm(input_data=input_data, also_update=True)
+
+        _check_error_for_data(v1_data, v2_data, error, error_norm_2)
```

Comment:
Besides checking that the error has the correct value, we also need to check that when `also_update=True` the internal variables where actually updated.

Reply:
@luisenp Did you mean to add a condition when `also_update=True`  and `also_update=False`?

Reply:
No, here I mean that when we pass `also_update=True` the objective should change the values of variables under the hood, and this unit test should confirm that this in fact happened.

Reply:
@luisenp can we do something like 
```
assert self.optim_vars["a1"] == input_data["a1"]

```

Reply:
Something like that, but for both "v1" and "v2".